### DISCUSSION TITLE
Which event is fired when the viewer is "stable"?

### DISCUSSION THREAD

**Discussion by alvaromartmart** [openseadragon/openseadragon#1980] [Q&A] [answered]
Hi,

I'd like to run some logic when the viewer is *stable*. By stable I mean that the `TileSource` has been loaded and the viewport has been drawn once. Is [`update-viewport`](https://openseadragon.github.io/docs/OpenSeadragon.Viewer.html#.event:update-viewport) what I'm looking for? The event is missing the documentation, I'm not sure it's the right one... any tips?

Thanks!

**Comment by iangilman** [comment, marked as answer]
You want the `fully-loaded-change` event. This fires on each image, so if you have more than one image in your viewer, you'll want to do something like is suggested in https://github.com/openseadragon/openseadragon/issues/1257.

**Reply by alvaromartmart** [reply]
Thank you @iangilman! That worked perfectly 😄 (I'm only working with one image at a time)

One thing though, in the documentation for that event it says
```
Fired when the TiledImage's "fully loaded" flag (whether all tiles necessary for this TiledImage to draw at the current view have been loaded) changes.
```
It wasn't clear to me from that description if the event is triggered *after* the TiledImage is actually **drawn** to the viewport, or when it's ready to be displayed -but just before.

**Reply by iangilman** [reply]
Good question! Upon examining the code, it appears that it's triggered after it's drawn (because the drawing process is when we learn whether we have everything we need). Feel free to add a mention of that to the docs (in a PR) if you want!
